### PR TITLE
mapbox-gl: add 'projection' property to MapboxOptions interface

### PR DESCRIPTION
**Pull request by lekoshimura**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://docs.mapbox.com/mapbox-gl-js/style-spec/projection/>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/mapbox-gl/index.d.ts [hunk 1]
```diff
@@ -771,6 +771,13 @@ declare namespace mapboxgl {
          */
         pitch?: number | undefined;
 
+        /**
+         * A style's projection property sets which projection a map is rendered in.
+         *
+         * @default 'mercator'
+         */
+         projection?: 'equalEarth' | 'albers' | 'mercator' | 'lambertConformalConic' | 'winkelTripel' | 'naturalEarth' | 'equirectangular';
```
Comment (gabritto) [inline types/mapbox-gl/index.d.ts:779]: I don't really understand this project/library, but the link to documentation you posted (https://docs.mapbox.com/mapbox-gl-js/style-spec/projection/) has an example of projection that looks like this:

```json
"projection": {
  "name": "albers",
  "center": [
    -154,
    50
  ],
  "parallels": [
    55,
    65
  ]
}
```
so isn't the type of the `projection` property here supposed to be a structured type? Like `projection?: { name: 'equalEarth' | 'albers' | ...., parallels?: ..., center?: ... }`?

[the remark this answers is not in the document]
Comment (lekoshimura) [reply types/mapbox-gl/index.d.ts]: Thanks for pointing this out, @gabritto!
I'm new here. Should I submit another PR? And leave this one here like this?

[the remark this answers is not in the document]
Comment (gabritto) [reply types/mapbox-gl/index.d.ts]: no, you should push new commits to this PR (as you did) :)

**Review by gabritto — APPROVED** [review]
thanks! please consider also updating the tests in future PRs

**Comment by lekoshimura** [comment]
Ready to merge

**Pull request merged**